### PR TITLE
TASK: Improve code / fix psalm issues

### PR DESCRIPTION
**Pull request by daniellienert**
Please have a look at the single commits where I documented what I have done and why.

### CODE REVIEW

File: Neos.Flow/Classes/I18n/Translator.php [hunk 1]
```diff
@@ -119,9 +119,11 @@ public function injectPluralsReader(Cldr\Reader\PluralsReader $pluralsReader)
      * @param string $sourceName Name of file with translations, base path is $packageKey/Resources/Private/Locale/Translations/
      * @param string $packageKey Key of the package containing the source file
      * @return string Translated $originalLabel or $originalLabel itself on failure
+     * @throws Exception\IndexOutOfBoundsException
+     * @throws Exception\InvalidFormatPlaceholderException
      * @api
      */
-    public function translateByOriginalLabel($originalLabel, array $arguments = [], $quantity = null, Locale $locale = null, $sourceName = 'Main', $packageKey = 'Neos.Flow')
+    public function translateByOriginalLabel($originalLabel, array $arguments = [], $quantity = null, Locale $locale = null, string $sourceName = 'Main', string $packageKey = 'Neos.Flow')
```
Comment (kdambekalns) [inline Neos.Flow/Classes/I18n/Translator.php]:
```suggestion
    public function translateByOriginalLabel(string $originalLabel, array $arguments = [], $quantity = null, Locale $locale = null, string $sourceName = 'Main', string $packageKey = 'Neos.Flow')
```

File: Neos.Flow/Classes/Package/PackageManager.php [hunk 2]
```diff
@@ -886,7 +886,7 @@ protected function getPackageKeyFromManifest(array $manifest, $packagePath): str
      * @param string $autoloadNamespace
      * @return string
      */
-    protected function derivePackageKey($composerName, $packageType = null, $packagePath = null, $autoloadNamespace = null): string
+    protected function derivePackageKey($composerName, $packageType = null, string $packagePath = '', $autoloadNamespace = null): string
```
Comment (kdambekalns) [inline Neos.Flow/Classes/Package/PackageManager.php]:
```suggestion
    protected function derivePackageKey(string $composerName, string $packageType = null, string $packagePath = '', string $autoloadNamespace = null): string
```

File: Neos.Flow/Classes/I18n/Translator.php (hunk 1 above)
Comment (daniellienert) [reply Neos.Flow/Classes/I18n/Translator.php]: Tried that, but there are tests, that fill it with null. So i reverted it again in https://github.com/neos/flow-development-collection/pull/1772/commits/c5d938b4e493c6c706226a41ed56c394e1a63419 and focused on getting it green.

**Comment by albe** [comment]
Sorry, I already started a review, then switched task and forgot about it. Looks good and thanks a lot Daniel! 🎉 for 6.0 being green again

**Pull request merged**